tools/gitignore/gitignore.py

```diff
@@ -60,10 +60,10 @@ def fnmatch_translate(pat):
             # TODO: this doesn't really handle invalid sequences in the right way
             if c == b"]":
                 seq = None
-                if parts[-1:] == b"[":
+                if parts[-1] == b"[":
                     parts = parts[:-1]
-                elif parts[-1:] == b"^" and parts[-2:-1] == b"[":
-                    parts = parts[:-2]
+                elif parts[-1] == b"^" and parts[-2] == b"[":
+                    raise ValueError
```

Comment:
Why is this an error rather than the previous intended behaviour of a no-op?

Reply:
Why did you change the expected test result to be an error: https://github.com/web-platform-tests/wpt/commit/6539f1e3433e77e23a33e293007f2da993c96abc#diff-a7ac75b33cb2a9ab48fca2210599111edf31888865c9ad7c85899248d40daca2

Reply:
Probably because there was a bug and I changed the expectation rather than noticing the bug :)

But I think in reality it doesn't matter how this case behaves, so an error is at least as good as handling it as a noop.